pkg/checks/checks.go

```diff
@@ -0,0 +1,249 @@
+		if !checkNamespace(ds.Namespace) {
+			continue
+		}
+		for {
+			ds, err := appclient.DaemonSets(ds.Namespace).Get(ds.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if ds.Status.NumberMisscheduled > 0 {
+				return fmt.Errorf("Daemonset[%v] in Namespace[%v] has missscheduled[%v]", ds.Name, ds.Namespace, ds.Status.NumberMisscheduled)
+			}
+
+			if ds.Status.DesiredNumberScheduled == ds.Status.CurrentNumberScheduled && ds.Status.DesiredNumberScheduled == ds.Status.NumberReady {
+				break
+			}
+			select {
+			case <-time.After(2 * time.Second):
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		}
+	}
+
+	// Statefulsets
+	ssList, err := appclient.StatefulSets("").List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, ss := range ssList.Items {
+		if !checkNamespace(ss.Namespace) {
+			continue
+		}
+		for {
+			ss, err := appclient.StatefulSets(ss.Namespace).Get(ss.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if ss.Status.Replicas == ss.Status.ReadyReplicas && ss.Status.ReadyReplicas == ss.Status.CurrentReplicas {
+				break
+			}
+			select {
+			case <-time.After(2 * time.Second):
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		}
+	}
+
+	// Deployments
+	dList, err := appclient.Deployments("").List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, d := range dList.Items {
+		if !checkNamespace(d.Namespace) {
+			continue
+		}
+		for {
+			d, err := appclient.Deployments(d.Namespace).Get(d.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if d.Status.UnavailableReplicas > 0 {
+				return fmt.Errorf("Deployment[%v] in namespace[%v] has unavailablereplicas[%v]", d.Name, d.Namespace, d.Status.UnavailableReplicas)
+			}
+			if d.Status.Replicas == d.Status.ReadyReplicas && d.Status.ReadyReplicas == d.Status.AvailableReplicas {
+				break
+			}
+			select {
+			case <-time.After(2 * time.Second):
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		}
+	}
+
+	// ReplicaSets
+	rsList, err := appclient.ReplicaSets("").List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, rs := range rsList.Items {
+		if !checkNamespace(rs.Namespace) {
+			continue
+		}
+		for {
+			rs, err := appclient.ReplicaSets(rs.Namespace).Get(rs.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if rs.Status.AvailableReplicas == 0 {
+				return fmt.Errorf("ReplicasSet[%v] in namespace[%v] has 0 availablereplicas", rs.Name, rs.Namespace)
+			}
+			if rs.Status.Replicas == rs.Status.ReadyReplicas && rs.Status.ReadyReplicas == rs.Status.AvailableReplicas {
+				break
+			}
+			select {
+			case <-time.After(2 * time.Second):
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		}
+	}
+
+	return nil
+}
+
+// WaitForPodsRunning verifies all pods are in running state
+func WaitForPodsRunning(ctx context.Context, coreclient *coreclient.CoreV1Client, appclient *appclient.AppsV1Client) error {
```

Comment:
not used - remove (save in a scratch pad in case you need it one day ;)

Reply:
done